app/src/main/java/com/example/healthroutineapplication/activities/ExerciseRoutineActivity.kt

```diff
@@ -0,0 +1,52 @@
+package com.example.healthroutineapplication.activities
+
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.widget.Toast
+import androidx.databinding.DataBindingUtil
+import com.example.healthroutineapplication.adapters.ExerciseRoutineAdapter
+import com.example.healthroutineapplication.R
+import com.example.healthroutineapplication.databinding.ActivityExerciseRoutineBinding
+import com.example.healthroutineapplication.models.ExerciseData
+
+class ExerciseRoutineActivity : AppCompatActivity() {
+    lateinit var binding: ActivityExerciseRoutineBinding
+    private var adapter = ExerciseRoutineAdapter(this)
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = DataBindingUtil.setContentView(this, R.layout.activity_exercise_routine)
+        binding.recyclerView.adapter = adapter
+        val routineTitle = intent.getStringExtra("exerciseRoutineTitle")
+        //루틴의 제목 intent로 받아옴
+        val dataList = intent.getSerializableExtra("exerciseRoutine") as ArrayList<ExerciseData>
+        //루틴의 운동들 ArrayList<ExerciseData> 받아옴
+        binding.exerciseRoutineTitle.text = routineTitle
+        adapter.setData(dataList)
+        binding.exerciseStartButton.setOnClickListener {
+            if (editCorrect(adapter.getData())) {
+                val intent = Intent(this, ExerciseStartingActivity::class.java)
+                intent.putExtra("exerciseStarting", adapter.getData())
+                startActivity(intent)
+                finish()
```

Comment:
early return 처리를 하면 else 조건이 필요 없어집니다.
```suggestion
                finish()
                return
          }
```